tests/testthat/test-renderer2-widerect.R

```diff
@@ -1,454 +1,428 @@
+  expect_true(old.year != new.year)
+})
+
+legend.td.xpath <-
+  '//tr[@class="region_variable"]//td[@class="legend_entry_label"]'
+rects_and_legends <- function(){
+  html <- getHTML()
+  list(rects=getNodeSet(html, '//rect[@id="United_States"]'),
+       legends=getStyleValue(html, legend.td.xpath, "opacity"))
+}
+test_that("clicking legend removes/adds countries", {
+  before <- rects_and_legends()
+  expect_equal(length(before$rects), 1)
+  expect_equal(sum(before$legends=="1"), 14)
+  expect_equal(sum(before$legends=="0.5"), 0)
+  clickID("plot_ts_region_variable_North_America")
+  Sys.sleep(1)
+  oneclick <- rects_and_legends()
+  expect_equal(length(oneclick$rects), 0)
+  expect_equal(sum(oneclick$legends=="1"), 12)
+  expect_equal(sum(oneclick$legends=="0.5"), 2)
+  clickID("plot_ts_region_variable_North_America")
+  Sys.sleep(1)
+  twoclicks <- rects_and_legends()
+  expect_equal(length(twoclicks$rects), 1)
+  expect_equal(sum(twoclicks$legends=="1"), 14)
+  expect_equal(sum(twoclicks$legends=="0.5"), 0)
+})
+
+clickID('updates_ms')
+ 
+
+
+test_that("pause stops animation (third time)", {
+  clickID("play_pause")
+  old.year <- getYear()
+  Sys.sleep(4)
+  new.year <- getYear()
+  expect_true(old.year == new.year)
+})
+
+# Function to send a key event
+sendKey <- function(key, code, keyCode) {
+  remDr$Input$dispatchKeyEvent(type = "keyDown", key = key, code = code, windowsVirtualKeyCode = keyCode, nativeVirtualKeyCode = keyCode)
+  remDr$Input$dispatchKeyEvent(type = "keyUp", key = key, code = code, windowsVirtualKeyCode = keyCode, nativeVirtualKeyCode = keyCode)
+}
+
+
+remDr$Runtime$evaluate("document.getElementsByClassName('show_hide_selector_widgets')[0].dispatchEvent(new CustomEvent('click'));")
+remDr$Runtime$evaluate("childDom = document.getElementsByClassName('year_variable_selector_widget')[0]; childDom.getElementsByClassName('selectize-input')[0].dispatchEvent(new CustomEvent('click'));")
+sendKey("Backspace", "Backspace", 8)
+remDr$Input$insertText(text = "1962")
+remDr$Runtime$evaluate("childDom = document.getElementsByClassName('year_variable_selector_widget')[0]; childDom.getElementsByClassName('selectize-input')[0].dispatchEvent(new CustomEvent('click'));") 
+sendKey("ArrowDown", "ArrowDown", 40)
+sendKey("Enter", "Enter", 13)
+   
+ 
+Sys.sleep(3)
+
+test_that("typing into selectize widget changes year to 1962", {
+  current.year <- getYear()
+  expect_identical(current.year, "1962")
+})
+
+
+
+
+getCountries <- function(){
+  country.labels <- getNodeSet(getHTML(), '//g[@class="geom8_text_ts"]//text')
+  sapply(country.labels, xmlValue)
+}
+
+test_that("initial countries same as first", {
+  country.vec <- getCountries()
+  expect_identical(sort(country.vec), sort(wb.facets$first$country))
+})
+
+
+remDr$Runtime$evaluate("childDom = document.getElementsByClassName('country_variable_selector_widget')[0]; childDom.getElementsByClassName('selectize-input')[0].dispatchEvent(new CustomEvent('click'));")
```

Comment:
same repeated JS code here as above, please make a helper function

Reply:
Add the JS code in clickSelector and used clickSelector function here